### PR TITLE
Calculating Sizes Dynamically

### PR DESCRIPTION
**Pull request by MapleSyyrup**
- calculate size dynamically

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -95,23 +103,35 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                      MediaQuery.of(context).padding.top) *
+                  0.4,
+              child: Chart(_recentTransactions),
+            ),
+            Container(
+                height: (MediaQuery.of(context).size.height -
+                        appBar.preferredSize.height -
+                        MediaQuery.of(context).padding.top) *
+                    0.6,
+                child: TransactionList(_userTransactions, _deleteTransaction)),
```
Comment (Nei1eveN) [inline lib/main.dart]: trailing commas

File: lib/widgets/chart.dart [hunk 2]
```diff
@@ -23,11 +23,14 @@ class Chart extends StatelessWidget {
         }
       }
 
+      print(DateFormat.E().format(weekDay),);
```
Comment (Nei1eveN) [inline lib/widgets/chart.dart]: remove comma if allowed.

File: lib/widgets/chart.dart [hunk 3]
```diff
@@ -40,26 +43,26 @@ class Chart extends StatelessWidget {
...
-                data['day'] ?? "",
-                data['amount'],
-                totalSpending == 0.0
-                    ? 0.0
-                    : (data['amount'] as double) / totalSpending,
-              ),
-            );
-          }).toList(),
+        elevation: 6,
+        margin: EdgeInsets.all(20),
```
Comment (Nei1eveN) [inline lib/widgets/chart.dart]: add `const` before `EdgeInsets`

File: lib/widgets/chart.dart [hunk 4]
```diff
@@ -40,26 +43,26 @@ class Chart extends StatelessWidget {
...
+        child: Padding(
+          padding: EdgeInsets.all(10),
```
Comment (Nei1eveN) [inline lib/widgets/chart.dart]: see above comment

File: lib/widgets/chart.dart [hunk 5]
```diff
@@ -40,26 +43,26 @@ class Chart extends StatelessWidget {
...
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: groupedTransactionValues.map((data) {
+              return Flexible(
+                fit: FlexFit.tight,
+                child: ChartBar(
+                  data['day'] ?? '',
+                  data['amount'],
+                  totalSpending == 0.0
+                      ? 0.0
+                      : (data['amount'] as double) / totalSpending,
```
Comment (Nei1eveN) [inline lib/widgets/chart.dart]: fix formatting. `totalSpending == 0.0 ? 0.0 : (data['amount'] as double) / totalSpending` seems to be more readable

**Review by Nei1eveN — CHANGES_REQUESTED** [review]
See my comments on each line.

File: lib/widgets/chart.dart (hunk 5 above)
Comment (MapleSyyrup) [reply lib/widgets/chart.dart]: Cannot change formatting. The refactoring tool automatically fixes the format like this:

 totalSpending == 0.0
                      ? 0.0
                      : (data['amount'] as double) / totalSpending,

File: lib/widgets/chart.dart (hunk 2 above)
Comment (Nei1eveN) [reply lib/widgets/chart.dart]: @MapleSyyrup can you remove the comma if allowed and will not cause any error in this line?

**Review by Nei1eveN — APPROVED** [review]

**Pull request merged**